mongodump_passthrough/functions.yml

```diff
@@ -0,0 +1,773 @@
+      add_expansions_to_env: true
+
+  "f_archive_credentials_setup_exec": &set_up_archive_credentials
+    command: subprocess.exec
+    params:
+      binary: "./src/mongosync/evergreen/archive_credentials_setup.sh"
+      env:
+        private_key_remote_bash_var: ${private_key_remote}
+      add_expansions_to_env: true
+
+  "get buildnumber": &get_buildnumber
+    command: keyval.inc
+    params:
+      key: "${build_variant}_v70"
+      destination: "builder_num"
+
+functions:
+  # Changed for mongodump_passthrough
+  "fetch source and install go":
+    # This clones mongo-tools and mongosync.
+    - command: git.get_project
+      params:
+        directory: src/github.com/mongodb/mongo-tools
+        revisions:
+          mongosync: *_mongosync_pin
+          migration-verifier: *_migration_verifier_pin
+    # The mongo-tools repo is cloned into src/github.com/mongodb/mongo-tools, but
+    # we also create a symbolic link at src/mongo-tools.
+    - *f_link_to_mongo_tools
+    - command: shell.exec
+      params:
+        shell: bash
+        script: |
+          ls -l src
+    # Make an evergreen expansion file with dynamic values
+    - command: subprocess.exec
+      params:
+        binary: "./src/mongosync/evergreen/generate_mongosync_expansion.sh"
+        add_expansions_to_env: true
+        env:
+          mongosync_dir: "src/mongosync"
+    # Load the expansion file to make an evergreen variable with the current unique version
+    - command: expansions.update
+      params:
+        file: src/mongosync/mongosync_expansion.yml
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: src/mongosync
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          MISE_INSTALL_PATH='${workdir}/.local/bin/mise' sh ./etc/mise.run.sh
+          mise settings experimental=true
+          # We only install Go here in order to get the system to a state where we can run mage. If
+          # more dev tools are needed, those tasks should run "mise install".
+          mise install go
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: src/mongosync
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          mise exec -- go version
+          mise exec -- go env
+
+  "run mise and npm install":
+    - command: shell.exec
+      params:
+        shell: bash
+        working_dir: src/mongosync
+        script: |
+          set -o errexit
+          set -o pipefail
+          set -o xtrace
+
+          ${PREPARE_SHELL}
+
+          mise install
+          mise exec -- npm install
+
+  # Added for mongodump_passthrough
+  #
+  # This build the mongo-tools executables, then copies mongogump and mongorestore
+  # into src/mongosync/dist.
+  "build mongodump and mongorestore":
```

Comment:
I don't understand why you're using `mise` to build the tools binaries. The mongo-tools repo is not using mise (yet?). So when we do an actual release of the tools, we'll just use whatever `go` is installed on the Evergreen host. I think we should use that same `go` for these tests.

Reply:
Changed so that we use the same setup as mongo-tools/common.yml "run make target" for code that runs in mongo-tools, but use the mise for code that lives in mongosync repo (the mongodump-suite-gen).

I did try to get everything to run without mise, but the shell setup (sourcing ./set_goenv.sh) gets confusing so I backed off to this.